control-plane/connect-inject/peering_acceptor_controller_test.go

```diff
@@ -503,9 +493,53 @@ func TestReconcile_CreateUpdatePeeringAcceptor(t *testing.T) {
 				}
 				return []*corev1.Secret{secret}
 			},
-			expectDeletedK8sSecret: &types.NamespacedName{
-				Name:      "some-old-secret",
-				Namespace: "default",
+			initialConsulPeerName: "acceptor-created",
+		},
+		{
+			name: "Peering exists in Consul, but secret doesn't and status is not set",
```

Comment:
My comment above is for this case-- and I think it'd mean adding an expShouldGenerate = true here

Reply:
I think `expShouldGenerate` one would be for the tests for `shouldGenerate` function. The behavior of that function actually didn't change (i just re-ordered some if conditions), so I didn't need to change the tests there. this test already asserts that the secret is created